src/compiler/program.ts

```diff
@@ -1160,6 +1206,75 @@ namespace ts {
 
         return program;
 
+        function collectCompilerExtensions(): ExtensionCollectionMap {
+            const extOptions = options.extensions;
+            const extensionNames = (extOptions instanceof Array) ? extOptions : getKeys(extOptions);
+            return groupBy(flatten(map(filter(map(extensionNames, name => {
+                let result: any;
+                let error: any;
+                if (host.loadExtension) {
+                    const resolved = resolveModuleName(name, combinePaths(currentDirectory, "tsconfig.json"), options, host, /*loadJs*/true).resolvedModule;
+                    if (resolved) {
+                        try {
+                            result = host.loadExtension(resolved.resolvedFileName);
+                        }
+                        catch (e) {
+                            error = e;
+                        }
+                    }
+                    else {
+                        error = new Error(`Host could not locate extension '${name}'.`);
+                    }
+                }
+                else {
+                    error = new Error("Extension loading not implemented in compiler host.");
+                }
+                if (error) {
+                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Extension_loading_failed_with_error_0, error.stack ? `${error}
+                    Stack trace:
+                    ${error.stack}` : error));
+                }
+                return {name, result, error};
+            }), res => !res.error), res => {
+                if (res.result) {

```

Comment:
Then why not make the filter on 1238 `res => res.result !== undefined`, or `res => !res.error && res.result !== undefined`?